### ISSUE TITLE
Question on use case

### ISSUE THREAD

**Issue by abitdodgy** [refile/refile#242]
I feel a bit sheepish asking because I haven't tried using Refile yet. I plan to try it tonight, and herein lies my question: I have three days to complete a prototype of a file upload feature that has several requirements, chief of which:

1. Multiple, direct uploads to S3. :+1: 
2. Access to files on S3. :+1: :question:  (A caveat: I need a CDN, and I can't access files directly from S3. As I understand Refile streams files directly to the app).
3. Files must be downloaded with original names. :+1: 
3. File versioning. :question: When a new upload is made, newly uploaded files must be looked up in the db by name. If they are found, they must be updated with the new files, and the old data must made into a file version, accessible if needed.

I haven't thought out the versioning feature too much, or if there is a better way to handle it. I can hazard a guess: When new uploads are made, lookup existing files by name, and transform existing files into versions, then let Refile finish it's upload.

I'm hoping someone can clue me in on if Refile is an ideal candidate, or if I have things terribly muddled up? I do hope so because it looks brilliant.

**Comment by jnicklas** [comment]
I cannot evaluate for you whether Refile is the right tool for the job, that choice is up to you ;)

If you need multiple uploads you are going to have to run off the master branch. If you need versioning, you're going to have to implement it yourself, Refile has nothing to assist you in this.

**Comment by abitdodgy** [comment]
@jnicklas thank you. I may have made my question long winded, and phrased it incorrectly. I would like to know if there is anything in my requirements that would prohibit me from using Refile (not being able to, say look up existing files, for example) or would be technically difficult or challenging to do in Refile. But that does not appear to be the case.

And a thank you for your work on Refile. Last year I did a project where I had to use jQuery FileUploads and bake my own direct S3 uploads. It was a nightmare.
